moto/sagemaker/models.py

```diff
@@ -3397,15 +3405,32 @@ def create_model_package(
         additional_inference_specifications: Any,
     ) -> str:
         model_package_version = None
-        if model_package_group_name is not None:
+        if model_package_group_name and model_package_name:
+            raise AWSValidationException(
+                "An error occurred (ValidationException) when calling the CreateModelPackage operation: Both ModelPackageName and ModelPackageGroupName are provided in the input. Cannot determine which one to use."
+            )
+        elif not model_package_group_name and not model_package_name:
+            raise AWSValidationException(
+                "An error ocurred (ValidationException) when calling the CreateModelPackag operation: Missing ARN."
+            )
+        elif model_package_group_name:
+            model_package_type = "Versioned"
+            model_package_name = model_package_group_name
             model_packages_for_group = [
                 x
                 for x in self.model_packages.values()
                 if x.model_package_group_name == model_package_group_name
             ]
+            if model_package_group_name not in self.model_package_groups:
+                raise AWSValidationException(
+                    "An error ocurred (ValidationException) when calling the CreateModelPackage operation: Model Package Group does not exist."
+                )
```

Comment:
This is also a thing to check, if the specified group doesn't exist, we can't create a ModelPackage for it.